marketdata/stream/subscription.go

```diff
@@ -177,7 +187,13 @@ func (c *client) handleSubChange(
 		return err
 	}
 
-	return <-request.result
+	select {
+	case err := <-request.result:
+		return err
+	case <-timeAfter(3 * time.Second):
+	}
+
+	return ErrSubscriptionChangeTimeout
```

Comment:
How would a user deal with this error? Would they retry the call? Wouldn't that result in a `ErrSubscriptionChangeAlreadyInProgress` error?

Reply:
Excellent note, thanks.